Lib/configparser.py

```diff
@@ -724,16 +724,25 @@ def read(self, filenames, encoding=None):
         filename may also be given.
 
         Return list of successfully read files.
+
+        Any exception other than an OSError raised while
+        parsing the file is propogated to the caller.
         """
         if isinstance(filenames, (str, bytes, os.PathLike)):
             filenames = [filenames]
         encoding = io.text_encoding(encoding)
         read_ok = []
         for filename in filenames:
             try:
-                with open(filename, encoding=encoding) as fp:
+                fp = open(filename, encoding=encoding)
+            except (OSError, ValueError):
+                continue
+            try:
+                with fp:
                     self._read(fp, filename)
             except OSError:
+                # Do not silence ValueError during the parsing
+                # phase such as those due to a wrong encoding.
```

Comment:
This block adds a good deal of complexity, doubling the number of exit points in the loop, adding a new variable, and creating several new blocks. I wonder if there's a way to extract this behavior into its own function.

Reply:
If on the other hand, we can remove the second try/except, the larger block remains much simpler, so no need to extract.